pkg/client/cache/listwatch.go

```diff
@@ -31,16 +32,16 @@ import (
 type ListerWatcher interface {
 	// List should return a list type object; the Items field will be extracted, and the
 	// ResourceVersion field will be used to start the watch in the right place.
-	List(options api.ListOptions) (runtime.Object, error)
+	List(options v1.ListOptions) (runtime.Object, error)
```

Comment:
Basically all of the discussions boil down to:

> Our internal code mode only works if the same object can be represented in all versions with no loss of data.  By definition, we try to avoid losing data.  When we do have that potential, we create new objects (ReplicaSet vs ReplicationController) and don't use object versioning.

So v1 is no *worse* than the internal version, except that right now we're introducing a change from internal to versioned that ripples out into other code before we're ready for it.

I'd rather preserve the internal api.ListOptions for now until we solve the label selector / field selector string representation in v1 (which causes other code changes) and do the conversion inside the list watcher impl.  The `v1` here shouldn't be core API v1 - it should be `meta.k8s.io/v1` and we don't have that yet.

We can always come back to this particular change.